### PR TITLE
deps: Updates OTEL collector dependencies to 0.52.0.

### PR DESCRIPTION
**Pull request by StefanKurek**
### Proposed Change
Updates OTEL collector dependencies to 0.52.0
Reworked plugin receiver because ConfigUnmarshaller was removed.
Fixed any other moved dependencies

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: exporter/googlecloudexporter/go.mod [hunk 1]
```diff
@@ -78,49 +76,60 @@ require (
...
 	github.com/sirupsen/logrus v1.8.1 // indirect
-	go.opentelemetry.io/collector/pdata v0.50.1-0.20220429151328-041f39835df7 // indirect
-	go.opentelemetry.io/collector/semconv v0.50.1-0.20220429151328-041f39835df7 // indirect
-	golang.org/x/sys v0.0.0-20220412211240-33da011f77ad // indirect
+	go.opentelemetry.io/collector/semconv v0.52.0 // indirect
+	golang.org/x/sys v0.0.0-20220503163025-988cb79eb6c6 // indirect
 )
 
-replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/googlecloudexporter v0.50.0 => github.com/observiq/opentelemetry-collector-contrib/exporter/googlecloudexporter v0.0.0-20220504030020-bb60df1ad6e0
-
```
Comment (BinaryFissionGames) [inline exporter/googlecloudexporter/go.mod:125]: Does upstream have logs support, or does that break here?

**Review by cpheps — CHANGES_REQUESTED** [review]
Looks like we missed updating the google cloud exporter.

File: go.mod [hunk 2]
```diff
@@ -412,14 +420,12 @@ require (
...
 	github.com/containerd/containerd v1.6.1 // indirect
-	github.com/klauspost/compress v1.15.1 // indirect
+	github.com/klauspost/compress v1.15.4 // indirect
 	github.com/shirou/gopsutil/v3 v3.22.4
 	github.com/spf13/cobra v1.4.0 // indirect
 	golang.org/x/sys v0.0.0-20220503163025-988cb79eb6c6
 )
 
-replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/googlecloudexporter v0.50.0 => github.com/observiq/opentelemetry-collector-contrib/exporter/googlecloudexporter v0.0.0-20220504030020-bb60df1ad6e0
-
```
Comment (BinaryFissionGames) [inline go.mod:422]: Same here; Does logs support still work for the googlecloudexporter?

File: exporter/googlecloudexporter/go.mod (hunk 1 above)
Comment (StefanKurek) [reply exporter/googlecloudexporter/go.mod]: fixed the dependency on the newest commits of our fork of the exporter

File: go.mod (hunk 2 above)
Comment (StefanKurek) [reply go.mod]: fixed the dependency on the newest commits of our fork of the exporter

**Review by BinaryFissionGames — APPROVED** [review]

**Review by jmwilliams89 — APPROVED** [review]

**Review by cpheps — APPROVED** [review]

**Pull request merged**